### PR TITLE
Catch up AppNetCsm test with latest changes

### PR DESCRIPTION
**Pull request by stanley-cheung**
#91, #95, #96, #97 should all be done to the AppNetCsm test as well, just that the test was merged after the fact.

Next step is to refactor out all the common logic into a Mixin class.

### CODE REVIEW

**Comment by stanley-cheung** [comment]
* [ ] [prod:grpc/core/master/linux/psm-csm](https://fusion2.corp.google.com/ci/kokoro/prod:grpc%2Fcore%2Fmaster%2Flinux%2Fpsm-csm/activity/77aa3d12-daf6-4f70-ba68-6f7f92bba0cc/tests)

**Review by sergiitk — COMMENTED** [review]
I'm ok with this change, but I think we'll benefit much more from moving common functionality into a mixin, similar to https://github.com/grpc/psm-interop/blob/main/framework/test_cases/session_affinity_mixin.py.

**Comment by sergiitk** [comment]
cc @ejona86 @zasweq @XuanWang-Amos

**Review by sergiitk — APPROVED** [review]

**Review by sergiitk — COMMENTED** [review]
Running ad-hoc test for the rest of the languages, and the oldest cpp branch.

**Review by sergiitk — CHANGES_REQUESTED** [review]
Go and Python failed. 

Go:
```py
Traceback (most recent call last):
  File "/tmp/tmp.6tlW3Jub2y/psm-interop/tests/app_net_csm_observability_test.py", line 331, in test_csm_observability
    self.assertDictEqual(
AssertionError: {'csm_mesh_id': <ANY>, 'csm_remote_workload_canonical_service': 'csm_canonical_service_name_from_server', 'csm_remote_workload_cluster_name': <ANY>, 'csm_remote_workload_location': <ANY>, 'csm_remote_workload_name': 'csm_workload_name_from_server', 'csm_remote_workload_namespace_name': 'psm-csm-server-20240606-2158-i8zi1', 'csm_remote_workload_project_id': 'grpc-testing', 'csm_remote_workload_type': 'gcp_kubernetes_engine', 'csm_service_name': 'psm-csm-backend-service-20240606-2158-i8zi1', 'csm_service_namespace_name': 'unknown', 'csm_workload_canonical_service': 'csm_canonical_service_name_from_client', 'grpc_method': 'grpc.testing.TestService/UnaryCall', 'grpc_status': 'OK', 'grpc_target': <ANY>, 'otel_scope_name': <ANY>, 'otel_scope_version': <ANY>, 'pod': 'psm-grpc-client-5888bfb897-xkl6x'} != {'csm_mesh_id': 'psm-csm-mesh-20240606-2158-i8zi1', 'csm_remote_workload_canonical_service': 'csm_canonical_service_name_from_server', 'csm_remote_workload_cluster_name': 'psm-interop-csm', 'csm_remote_workload_location': 'us-central1', 'csm_remote_workload_name': 'csm_workload_name_from_server', 'csm_remote_workload_namespace_name': 'psm-csm-server-20240606-2158-i8zi1', 'csm_remote_workload_project_id': 'grpc-testing', 'csm_remote_workload_type': 'gcp_kubernetes_engine', 'csm_service_name': 'psm-csm-backend-service-20240606-2158-i8zi1', 'csm_workload_canonical_service': 'csm_canonical_service_name_from_client', 'grpc_method': 'grpc.testing.TestService/UnaryCall', 'grpc_status': 'OK', 'grpc_target': 'xds:///psm-grpc-server:58628', 'otel_scope_name': 'grpc-go', 'otel_scope_version': '1.65.0-dev', 'pod': 'psm-grpc-client-5888bfb897-xkl6x'}
Missing entries:
'csm_service_namespace_name': 'unknown'
```

Python:
```py
Traceback (most recent call last):
  File "/tmp/tmp.xm4HFPozRO/psm-interop/framework/xds_k8s_testcase.py", line 879, in tearDown
    self.assertEqual(
AssertionError: 5 != 0 : Server container unexpectedly restarted 5 times during test. In most cases, this is caused by the test client app crash.
```

File: tests/app_net_csm_observability_test.py [hunk 1]
```diff
@@ -175,6 +176,8 @@ class AppNetCsmObservabilityTest(xds_k8s_testcase.AppNetXdsKubernetesTestCase):
     def is_supported(config: skips.TestConfig) -> bool:
         if config.client_lang == _Lang.CPP:
             return config.version_gte("v1.62.x")
+        elif config.client_lang in _Lang.GO | _Lang.JAVA | _Lang.PYTHON:
```
[the remark this answers is not in the document]
Comment (sergiitk) [reply tests/app_net_csm_observability_test.py:179]: Should've used more recent master commit for the python job. I used the overnight one (grpc/grpc@60301d40e0), which didn't include grpc/grpc#36827.

- [ ] [grpc/core/master/linux/psm-csm-python](https://source.cloud.google.com/results/invocations/f8652383-b45e-4818-a147-a05d28ffd146)

[the remark this answers is not in the document]
Comment (sergiitk) [reply tests/app_net_csm_observability_test.py:179]: Re-testing go using @zasweq's fix: grpc/grpc-go#7309.

- [ ] [grpc/go/master/branch/psm-csm](https://source.cloud.google.com/results/invocations/ca893e8b-9a62-474a-b0aa-5be1d015eae0)

**Review by sergiitk — APPROVED** [review]
Go fix  merged: grpc/grpc#36827.

**Pull request merged**